modules/seo.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Module Name: seo
+ * Module Description: Adds tools to enhance your site's content for better results on search engines and social media.
+ * Sort Order: 35
+ * First Introduced: 4.3
+ * Requires Connection: No
+ * Auto Activate: Yes
+ * Module Tags: Social, Appearance
+ */
+
+include dirname( __FILE__ ) . '/seo/helpers-seo.php';
+include dirname( __FILE__ ) . '/seo/title-seo.php';
+include dirname( __FILE__ ) . '/seo/posts-seo.php';
+
+/**
+ * An SEO expert walks into a bar, bars, pub, public house, Irish pub, drinks, beer, wine, liquor, Grey Goose, Cristal...
+ */
+class SEO {
+	function __construct() {
+		add_action( 'init', array( $this, 'init' ) );
+	}
+
+	function init() {
+		add_action( 'wp_head', array( $this, 'meta_tags' ) );
+
+		// Add support for editing page excerpts in pages, regardless of theme support.
+		add_post_type_support( 'page', 'excerpt' );
+
+		// Overwrite page title with custom SEO meta title for themes that support title-tag
+		add_filter( 'pre_get_document_title', 'get_custom_title' );
+
+		// Add overwrite support for themes that don't support title-tag
+		add_filter( 'wp_title', 'get_custom_title' );
+	}
+
+	/**
+	 * "The first major crawler-based search engines to use the meta keywords tag were Infoseek and AltaVista"
+	 *
+	 * Source: http://searchenginewatch.com/article/2066825/Death-Of-A-Meta-Tag
+	 */
+	function meta_tags() {
+
+		global $wp_query;
+
+		$period = $template = '';
+
+		$description = get_front_page_meta_description() ?: get_bloginfo( 'description' );
+
+		$meta = array(
+			'title'       => sprintf( _x( '%1$s on %2$s', 'Blog Title on WordPress.com' ), get_bloginfo( 'title' ), 'WordPress.com' ),
```

Comment:
Yes, there is a lot of stuff in `seo.php` that will require modification or complete removal (mentioned in tasks in PR description). Only basic modifications were made when compared to wpcom file in order to get the basic Adv. SEO features working, so we can start testing and modifying Calypso and wpcom side in parallel. This is a very early stage WIP version that I submitted to make it easier for my teammates to review/contribute/test. :slightly_smiling_face: